src/ucs/sys/sock.c

```diff
@@ -14,15 +14,33 @@
 #include <unistd.h>
 #include <errno.h>
 #include <string.h>
+#include <sys/uio.h>
+/* need this to get IOV_MAX on some platforms. */
+#ifndef __need_IOV_MAX
+#define __need_IOV_MAX
+#endif
 #include <limits.h>
 
 
 #define UCS_SOCKET_MAX_CONN_PATH "/proc/sys/net/core/somaxconn"
 
+/* Maximum length of the 'struct iovec' array in a single call to
+ * readv/writev/sendmsg/recvmsg/etc. */
+#if defined(IOV_MAX)
+#define UCS_SOCKET_MAX_IOV       IOV_MAX
```

Comment:
can we avoid defining UCS_SOCKET_MAX_IOV macro and move this code inside ucs_socket_max_iov()?
in order to discourage anyone from using UCS_SOCKET_MAX_IOV directly and only rely on  ucs_socket_max_iov()

Reply:
yes, good idead
done